pkg/await/awaiters.go

```diff
@@ -448,33 +577,41 @@ func untilCoreV1ServiceAccountInitialized(
 
 // --------------------------------------------------------------------------
 
-func untilExtensionsV1Beta1IngressInitialized(
-	clientForResource, clientForEvents dynamic.ResourceInterface, obj *unstructured.Unstructured,
-) error {
+func untilExtensionsV1Beta1IngressInitialized(c initAwaitConfig) error {
+	clientForEvents, err := client.FromGVK(c.pool, c.disco, schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Event",
+	}, c.currentInputs.GetNamespace())
```

Comment:
Perhaps minor - should this be a helper function defined on `initAwaitConfig` that can be reused in the few places it is currently inlined?